src/components/vice/admin/table/index.js

```diff
@@ -0,0 +1,281 @@
+    const rowID = id(baseID, "row");
+    const collapseID = id(rowID, "collapse");
+
+    return (
+        <>
+            <TableRow className={classes.row} key={rowID} id={rowID}>
+                <TableCell key={expanderID} id={expanderID}>
+                    <IconButton
+                        aria-label="expand row"
+                        size="small"
+                        onClick={() => setOpen(!open)}
+                    >
+                        {open ? <KeyboardArrowUp /> : <KeyboardArrowDown />}
+                    </IconButton>
+                </TableCell>
+
+                {columns.slice(startColumn, endColumn).map((column) => {
+                    const fieldID = id(rowID, column.field);
+                    return (
+                        <TableCell key={fieldID} id={fieldID}>
+                            {row[column.field]}
+                        </TableCell>
+                    );
+                })}
+            </TableRow>
+
+            <TableRow key={collapseID} id={collapseID}>
+                <TableCell
+                    style={{ paddingBottom: 0, paddingTop: 0, width: "90%" }}
+                    colSpan={endColumn}
+                >
+                    <Collapse in={open} timeout="auto" unmountOnExit>
+                        <ExtendedDataCard
+                            columns={columns.slice(endColumn)}
+                            row={row}
+                            collapseID={collapseID}
+                        />
+                    </Collapse>
+                </TableCell>
+            </TableRow>
+        </>
+    );
+};
+
+const CollapsibleTable = ({ columns, rows, title }) => {
+    const classes = useStyles();
+    const theme = useTheme();
+    const isSmall = useMediaQuery(theme.breakpoints.up("xs"));
+    const isMedium = useMediaQuery(theme.breakpoints.up("sm"));
+    const isLarge = useMediaQuery(theme.breakpoints.up("lg"));
+    const isXL = useMediaQuery(theme.breakpoints.up("xl"));
+
+    let startColumn;
+    let endColumn;
+
+    if (isXL) {
+        startColumn = 1;
+        endColumn = 7;
+    } else if (isLarge) {
+        startColumn = 1;
+        endColumn = 6;
+    } else if (isMedium) {
+        startColumn = 1;
+        endColumn = 4;
+    } else if (isSmall) {
+        startColumn = 1;
+        endColumn = 2;
+    } else {
+        startColumn = 1;
+        endColumn = 7;
+    }
+
+    // The first entry in columns should be the expander columns,
+    // so default to the second entry for sorting. The field is the
+    // actual name of the column.
+    const [orderColumn, setOrderColumn] = useState(columns[1].field);
+
+    const [order, setOrder] = useState("asc");
+
+    const tableID = id(ids.ROOT);
+
+    const sortAscending = (one, two) =>
+        one[orderColumn].localeCompare(two[orderColumn]) * -1;
+
+    const sortDescending = (one, two) =>
+        one[orderColumn].localeCompare(two[orderColumn]);
```

Comment:
We also have [these](https://github.com/cyverse-de/ui-lib/blob/master/src/util/table/TableSort.js) helper methods in ui-lib.  They're organized and labeled as being for Tables, but that's only because that's the MUI component we copied that code from 😄

Reply:
I'll look into this in a follow-up PR, since I may need to add pagination anyway. I can make all those changes at once.